### DISCUSSION TITLE
GnosisSafe support

### DISCUSSION THREAD

**Discussion by platschi** [rainbow-me/rainbowkit#711] [Q&A]
Hey, looking at issue #348 (which is closed) there's quite some interest in adding GnosisSafe support to rainbowkit. This seems essential as *many* projects use GnosisSafe to connect to various dApps, and a major blocker for many to integrate rainbowkit.

Would love to know if this is something coming rather sooner than later.

**Comment by jacobedawson** [comment]
@platschi and for anyone looking to integrate Gnosis Safe into their dapp, you can actually achieve this with wagmi while having Rainbow manage standard wallets e.g. MM, Rainbow, Coinbase, Walletconnect, etc. 

Gnosis (safe-global) have a Safe Apps Wagmi connector which includes examples, you can find it here: 

Package: https://github.com/safe-global/safe-apps-sdk/tree/master/packages/safe-apps-wagmi
Examples: https://github.com/safe-global/safe-apps-sdk/tree/master/examples/wagmi

I did notice a couple of things in the safe-global documentation that weren't immediately obvious, thankfully a PR created by Olympus DAO demonstrates a couple of extra pieces of code that make the process easier to understand & implement: 
https://github.com/OlympusDAO/olympus-frontend/pull/2164
https://github.com/OlympusDAO/olympus-frontend/pull/2164/files

Essentially, you will not see a button inside the Rainbow modal that says "Gnosis Safe" to initially log in (nor a modal or connect button at all), since you will create an autoconnect hook that recognizes when the dapp is loaded inside the Safe dapp - **this hook will automatically connect your dapp to the Safe multisig address**. The mental model needed to understand this is slightly different to the traditional dapp / wallet process. If you were to log out, however, you will see a Gnosis button in the Rainbow modal. 

![image](https://user-images.githubusercontent.com/4788506/190397475-13bb531a-fa75-4ee9-ad81-caaf98dbfa2b.png)

In order for transactions to be signed, ofc you must be signed into the Safe dapp, which is something you would do without invoking Rainbowkit. Rainbow might decide to add a convenience method down the track that simplifies the implementation for us, but if it's the only blocker for you then the above code / examples should help. 

Feel free to mention me here if you have any questions, I'll try to help. 

Cheers, 

Jake
